src/Testing/LeanCode.IntegrationTestHelpers/ConfigurationOverrides.cs

```diff
@@ -6,89 +6,71 @@ namespace LeanCode.IntegrationTestHelpers;
-        this.connectionStringKey = connectionStringKey ?? ConnectionStringKeyDefault;
-        this.internalBaseKey = internalBaseKey ?? InternalBaseKeyDefault;
-        this.publicBaseKey = publicBaseKey ?? PublicBaseKeyDefault;
+        this.data = data;
     }
 
+    public static OverridesBuilder CreateBuilder() => new();
+
     public IConfigurationProvider Build(IConfigurationBuilder builder)
     {
-        return new Provider(
-            minimumLevel,
-            enableInternalLogs,
-            connectionStringBase,
-            connectionStringKey,
-            internalBaseKey,
-            publicBaseKey
-        );
+        return new Provider(data);
     }
 
     private sealed class Provider : ConfigurationProvider
     {
-        private readonly LogEventLevel minimumLevel;
-        private readonly bool enableInternalLogs;
-        private readonly string connectionStringBase;
-        private readonly string connectionStringKey;
-        private readonly string internalBaseKey;
-        private readonly string publicBaseKey;
+        private readonly Dictionary<string, string?> data;
 
-        public Provider(
-            LogEventLevel minimumLevel,
-            bool enableInternalLogs,
-            string connectionStringBase,
-            string connectionStringKey,
-            string internalBaseKey,
-            string publicBaseKey
-        )
+        public Provider(Dictionary<string, string?> data)
         {
-            this.minimumLevel = minimumLevel;
-            this.enableInternalLogs = enableInternalLogs;
-            this.connectionStringBase = connectionStringBase;
-            this.connectionStringKey = connectionStringKey;
-            this.internalBaseKey = internalBaseKey;
-            this.publicBaseKey = publicBaseKey;
+            this.data = data;
         }
 
         public override void Load()
         {
-            var dbName = $"integration_tests_{Guid.NewGuid():N}";
-            var rest = Environment.GetEnvironmentVariable(connectionStringBase);
-            var dbConnStr = $"Database={dbName};" + rest;
+            Data = data;
+        }
+    }
+
+    public static string OverrideDatabaseConnectionString(
+        string dbPrefix,
+        string sourceEnv = ConnectionStringBaseDefault,
+        string varName = "Database"
+    )
+    {
+        var dbName = $"{dbPrefix}_{Guid.NewGuid():N}";
+        var rest = Environment.GetEnvironmentVariable(sourceEnv);
+        return $"{varName}={dbName};" + rest;
+    }
 
-            Data = new Dictionary<string, string?>
-            {
-                [connectionStringKey] = dbConnStr,
-                [internalBaseKey] = "http://localhost",
-                [publicBaseKey] = "http://localhost",
-                [IHostBuilderExtensions.EnableDetailedInternalLogsKey] = enableInternalLogs.ToString(),
-                [IHostBuilderExtensions.MinimumLogLevelKey] = minimumLevel.ToString(),
-            };
+    public class OverridesBuilder
```

Comment:
We can simplify this IMO - the `ConfigurationOverrides` and `OverridesBuilder` can be one and the same class. The overrides class can work as a builder (it is a builder - for the `Provider` class) and we don't need to make one more indirection.

Otherwise, looks good.